cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst

```diff
@@ -1600,6 +1600,202 @@ More information about workflow endpoint can be found at :ref:`Workflows <http-r
        - | ``[{"runid":"cad83d45-ecfb-4bf8-8cdb-4928a5601b0e","start":1415051892,"end":1415057103,"status":"STOPPED"}]``
 
 
+Retrieving Run Records
+----------------------
+
+To retrieve the latest run records for various applications and different program types, using an HTTP POST method::
+
+  GET /v3/namespaces/<namespace-id>/runs
+
+.. list-table::
+   :widths: 20 80
+   :header-rows: 1
+
+   * - Parameter
+     - Description
+   * - ``namespace-id``
+     - Namespace ID
+
+with a JSON array in the request body consisting of multiple JSON objects with these parameters:
+
+.. list-table::
+   :widths: 20 80
+   :header-rows: 1
+
+   * - Parameter
+     - Description
+   * - ``appId``
+     - Name of the application being called
+   * - ``programType``
+     - One of ``mapreduce``, ``spark``, ``workflow``, ``flow``, ``service``, or ``worker``
+   * - ``programId``
+     - Name of the program (*mapreduce*, *spark*, *workflow*, *flow*, *service*, or *worker*) being called
+
+The response will be an array of JsonObjects each of which will contain the three input parameters
+as well as 2 fields, "runs" which is a list of the latest run records and "statusCode" which maps to the
```

Comment:
as well as two of three possible extra fields -- "runs", which is a list of the latest run record for that program, "statusCode", which maps to the status code for retrieving the runs for that program, and "error" if there was an error retrieving runs for that program. The "statusCode" property will always be included, but "runs" and "error" are mutually exclusive.